Firestore/tests/Unit/FirestoreSessionHandlerTest.php

```diff
@@ -0,0 +1,394 @@
+        $this->collection->document(self::SESSION_NAME . ':sessionid')
+            ->shouldBeCalledTimes(1)
+            ->willReturn($docRef);
+        $firestoreSessionHandler = new FirestoreSessionHandler(
+            $this->firestore->reveal()
+        );
+        $firestoreSessionHandler->open(self::SESSION_SAVE_PATH, self::SESSION_NAME);
+        $ret = $firestoreSessionHandler->read('sessionid');
+
+        $this->assertEquals('', $ret);
+    }
+
+    /**
+     * @expectedException PHPUnit_Framework_Error_Warning
+     */
+    public function testReadWithException()
+    {
+        $this->firestore->collection(self::SESSION_SAVE_PATH)
+            ->shouldBeCalledTimes(1)
+            ->willReturn($this->collection->reveal());
+        $firestoreSessionHandler = new FirestoreSessionHandler(
+            $this->firestore->reveal()
+        );
+        $this->collection->document(self::SESSION_NAME . ':sessionid')
+            ->shouldBeCalledTimes(1)
+            ->willThrow((new Exception()));
+
+        $firestoreSessionHandler->open(self::SESSION_SAVE_PATH, self::SESSION_NAME);
+        $ret = $firestoreSessionHandler->read('sessionid');
+
+        $this->assertEquals('', $ret);
+    }
+
+    public function testReadEntity()
+    {
+        $id = self::SESSION_NAME . ':sessionid';
+        $snapshot = $this->prophesize(DocumentSnapshot::class);
+        $snapshot->exists()
+            ->shouldBeCalledTimes(1)
+            ->willReturn(true);
+        $snapshot->offsetExists('data')
+            ->shouldBeCalledTimes(1)
+            ->willReturn(true);
+        $snapshot->get('data')
+            ->shouldBeCalledTimes(1)
+            ->willReturn('sessiondata');
+        $docRef = $this->prophesize(DocumentReference::class);
+        $docRef->snapshot()
+            ->shouldBeCalledTimes(1)
+            ->willReturn($snapshot);
+        $this->collection->document($id)
+            ->shouldBeCalledTimes(1)
+            ->willReturn($docRef);
+        $this->firestore->collection(self::SESSION_SAVE_PATH)
+            ->shouldBeCalledTimes(1)
+            ->willReturn($this->collection->reveal());
+        $firestoreSessionHandler = new FirestoreSessionHandler(
+            $this->firestore->reveal()
+        );
+        $firestoreSessionHandler->open(self::SESSION_SAVE_PATH, self::SESSION_NAME);
+        $ret = $firestoreSessionHandler->read('sessionid');
+
+        $this->assertEquals('sessiondata', $ret);
+    }
+
+    public function testWrite()
+    {
+        $id = self::SESSION_NAME . ':sessionid';
+        $docRef = $this->prophesize(DocumentReference::class);
+        $this->firestore->collection(self::SESSION_SAVE_PATH)
+            ->shouldBeCalledTimes(1)
+            ->willReturn($this->collection->reveal());
+        $phpunit = $this;
+        $this->firestore->runTransaction(Argument::type('closure'))
+            ->shouldBeCalledTimes(1)
+            ->will(function($args) use ($phpunit, $docRef) {
```

Comment:
To fix the phpcs error in kokoro:

```suggestion
            ->will(function ($args) use ($phpunit, $docRef) {
```